gammapy/datasets/flux_points.py

```diff
@@ -275,107 +296,102 @@ def plot_residuals(self, ax=None, method="diff", **kwargs):
+        ax.axhline(0, color=kwargs["color"], lw=0.5)
 
         # format axes
-        ax.axhline(0, color="black", lw=0.5)
-        ax.set_ylabel("Residuals {}".format(unit.__str__()))
-        ax.set_xlabel(f"Energy ({self._e_unit})")
+        ax.set_xlabel(f"Energy [{self._e_unit}]")
         ax.set_xscale("log")
         ax.set_xlim(self._e_range.to_value(self._e_unit))
-        y_max = 2 * np.nanmax(residuals).value
-        ax.set_ylim(-y_max, y_max)
+        label = self._residuals_labels[method]
+        ax.set_ylabel(f"Residuals ({label}){f' [{unit}]' if unit else ''}")
+        ymin = 1.05 * np.nanmin(residuals.value - yerr[0])
+        ymax = 1.05 * np.nanmax(residuals.value + yerr[1])
+        ax.set_ylim(ymin, ymax)
         return ax
 
-    def plot_spectrum(self, ax=None, fp_kwargs=None, model_kwargs=None):
-        """
-        Plot spectrum including flux points and model.
+    def plot_spectrum(self, ax=None, fp_kwargs=None, model_kwargs=None, **kwargs):
+        """Plot spectrum including flux points and model.
 
         Parameters
         ----------
-        ax : `~matplotlib.pyplot.Axes`
-            Axes object.
+        ax : `~matplotlib.axes.Axes`
+            Axes to plot on.
         fp_kwargs : dict
-            Keyword arguments passed to `FluxPoints.plot`.
+            Keyword arguments passed to `gammapy.estimators.FluxPoints.plot`.
         model_kwargs : dict
-            Keywords passed to `SpectralModel.plot` and `SpectralModel.plot_error`
+            Keyword arguments passed to `gammapy.modeling.models.SpectralModel.plot` and
+            `gammapy.modeling.models.SpectralModel.plot_error`.
+        **kwargs: dict
+            Keyword arguments passed to all plot methods.
 
         Returns
         -------
-        ax : `~matplotlib.pyplot.Axes`
+        ax : `~matplotlib.axes.Axes`
             Axes object.
         """
-        import matplotlib.pyplot as plt
-
-        ax = plt.gca() if ax is None else ax
-        fp_kwargs = {} if fp_kwargs is None else fp_kwargs
-        model_kwargs = {} if model_kwargs is None else model_kwargs
+        fp_kwargs = fp_kwargs or {}
+        model_kwargs = model_kwargs or {}
 
-        kwargs = {
-            "flux_unit": "erg-1 cm-2 s-1",
-            "energy_unit": "TeV",
-            "energy_power": 2,
-        }
+        kwargs.setdefault("energy_power", 2)
+        kwargs.setdefault("energy_unit", "TeV")
+        kwargs.setdefault("flux_unit", "erg-1 cm-2 s-1")
 
         # plot flux points
         plot_kwargs = kwargs.copy()
         plot_kwargs.update(fp_kwargs)
         plot_kwargs.setdefault("label", "Flux points")
-        ax = self.data.plot(ax=ax, **plot_kwargs)
+        ax = self.data.plot(ax, **plot_kwargs)
 
         plot_kwargs = kwargs.copy()
-        plot_kwargs.setdefault("energy_range", self._e_range)
-        plot_kwargs.setdefault("zorder", 10)
         plot_kwargs.update(model_kwargs)
+        plot_kwargs.setdefault("energy_range", self._e_range)
```

Comment:
the reco or true energy range? @adonath, as we have discussed in the past, is it not useful to precise it always in the variable names?